packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -78,136 +76,20 @@ export async function prefetchLatestSnapshot(
                 if (tokenFetchOptions.refresh) {
                     attempts = 2;
                 }
-                const storageToken = await storageTokenFetcher(tokenFetchOptions, "PrefetchLatestSnapshot");
-                assert(storageToken !== null, "Storage token should not be null");
-                return fetchSnapshot(
-                    snapshotUrl,
+                await fetchLatestSnapshotCore(
                     odspResolvedUrl,
+                    storageTokenFetcher,
+                    tokenFetchOptions,
                     hostSnapshotFetchOptions,
-                    storageToken,
                     odspLogger,
-                    persistedCache,
-                    snapshotCacheEntry,
+                    snapshotDownloader,
+                    putInCache,
                 );
+                let cached = false;
+                await cacheP?.then(() => cached = true);
+                return cached;
```

Comment:
I think this would be same and more readable:
```typescript
await cacheP;
return true;
```

Reply:
Actually, there is a difference, but I think we want to assert it.
I.e.
```typescript
assert(cacheP !== undefined);
await cacheP;
return true;
```
I.e. use exceptions as the only mechanism to communicate errors (as host needs to catch and handle exceptions anyway, no exception means we successfully pre-cached).